app/seeders/basic_data/setting_seeder.rb

```diff
@@ -54,8 +54,10 @@ def data
       # deviate from the defaults specified in settings.yml here
       # to set a default role. The role cannot be specified in the settings.yml as
       # that would mean to know the ID upfront.
-      default_role_id = Role.find_by(name: I18n.t(:default_role_project_admin)).id
-      settings['new_project_user_role_id'] = default_role_id
+      if Setting.new_project_user_role_id.empty?
```

Comment:
This should rather be
`Setting.where(name: 'new_project_user_role_id').exists?` because
`Setting.new_project_user_role_id.empty?` is also true when the value `""` is stored in the DB. It would thus overwrite settings where users have chosen to not have a default role.

Alternatively, you could take the `settings_not_in_db` method again to check. That would be more in line with the rest of the implementation.